packages/grid/src/mouse-handlers/GridLinkMouseHandler.ts

```diff
@@ -0,0 +1,256 @@
+        x >= left && x <= right && y >= top && y <= bottom;
+
+      if (cursorIsInBounds) {
+        // Reset font
+        context.restore();
+
+        this.currentLinkBox = {
+          left,
+          top,
+          right,
+          bottom,
+          width: linkWidth,
+          height: linkHeight,
+          value,
+          href,
+        };
+
+        return this.currentLinkBox;
+      }
+    }
+
+    // If it reaches this point, then the mouse is not hovering a link so restore context and currentLinkBox
+    context.restore();
+    this.currentLinkBox = undefined;
+
+    return null;
+  }
+
+  /**
+   * Set the cursor to pointer when hovering of a link
+   * @param gridPoint Mouse position information
+   * @param grid The grid
+   * @returns False
+   */
+  private setCursor(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    if (GridLinkMouseHandler.getHoveringLinkInfo(gridPoint, grid) != null) {
+      this.cursor = 'pointer';
+    } else {
+      this.cursor = null;
+    }
+    return false;
+  }
+
+  onMove(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    return this.setCursor(gridPoint, grid);
+  }
+
+  onDown(
+    gridPoint: GridPoint,
+    grid: Grid,
+    event: GridMouseEvent
+  ): EventHandlerResult {
+    this.isDown = true;
+    const link = GridLinkMouseHandler.getHoveringLinkInfo(gridPoint, grid);
+
+    // If a modifier key or shift is down, we don't want to handle it and set isDown to false so onUp won't handle it either
+    if (GridUtils.isModifierKeyDown(event) || event.shiftKey || link == null) {
+      this.isDown = false;
+      return false;
+    }
+
+    if (this.timeoutId) {
+      clearTimeout(this.timeoutId);
+    }
+    this.isHold = false;
+    this.timeoutId = setTimeout(() => {
+      this.isHold = true;
+      const { column, row } = gridPoint;
+      const { model } = grid.props;
+
+      // After 1 second, select the row and if it is an input table, start editing
+      grid.clearSelectedRanges();
+      grid.focus();
+      grid.moveCursorToPosition(column, row);
+      if (
+        isEditableGridModel(model) &&
+        column != null &&
+        row != null &&
+        model.isEditableRange(GridRange.makeCell(column, row))
+      ) {
+        grid.startEditing(column, row);
+      }
+    }, 1000);
+
+    return true;
+  }
+
+  onUp(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    const link = GridLinkMouseHandler.getHoveringLinkInfo(gridPoint, grid);
+    if (link == null || !this.isDown) {
+      this.isDown = false;
+      this.isHold = false;
+      return false;
+    }
+
+    // Link is not null, isDown is true
+    // If isHold is true, then the select already happened so onUp shouldn't do anything
+    if (!this.isHold) {
+      // Cancel the timeout and open a new tab with the link
+      if (this.timeoutId != null) {
+        clearTimeout(this.timeoutId);
+      }
+      this.timeoutId = undefined;
+      window.open(link.href, '_blank');
```

Comment:
Instead of calling `window.open` directly, we should have an `onTokenClicked` prop set on `Grid` that gets called with the click info and the token.
Then `Grid` can have a default value for that prop that calls `window.open` with the token if it's a `LinkToken`.

Also, we should pass the `noopener` and `noreferrer` window options, since we don't know where this link is going: https://developer.mozilla.org/en-US/docs/Web/API/Window/open
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/noopener
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/noreferrer